apps/builder/app/builder/features/seo/project-redirects.tsx

```diff
@@ -0,0 +1,170 @@
+import {
+  Grid,
+  Separator,
+  Label,
+  Flex,
+  InputField,
+  Button,
+  Text,
+  theme,
+  List,
+  ListItem,
+  css,
+  IconButton,
+} from "@webstudio-is/design-system";
+import { DeleteIcon } from "@webstudio-is/icons";
+import { useState, type ChangeEvent, useEffect } from "react";
+import type { ProjectSettings } from "./project-settings";
+
+const redirectListStyle = css({
+  maxHeight: theme.spacing[22],
+  paddingLeft: 0,
+  overflowY: "scroll",
+  listStyle: "none",
+});
+
+export const ProjectRedirectionSettings = (props: {
+  settings: ProjectSettings;
+  onSettingsChange: (settings: ProjectSettings) => void;
+}) => {
+  const [oldPath, setOldPath] = useState<string | null>(null);
+  const [newPath, setNewPath] = useState<string | null>(null);
+  const [isValidOldPath, setIsValidOldPath] = useState<boolean>(false);
+  const [isValidNewPath, setIsValidNewPath] = useState<boolean>(false);
+
+  const redirects = props.settings?.redirects ?? {};
+  const redirectKeys = Object.keys(redirects);
+  const isValidRedirects =
+    oldPath !== null && isValidOldPath && newPath !== null && isValidNewPath;
+
+  useEffect(() => {
+    if (oldPath === null) {
+      return setIsValidOldPath(true);
+    }
+    return setIsValidOldPath(oldPath.charAt(0) === "/");
+  }, [oldPath]);
```

Comment:
I think the usage of useEffect is "wrong" here, you can call setIsValidOldPath from event handler, I think this is a classic example from react docs "how not to do it" because useEffect is meant for synchronization of states, not really to do effects

Reply:
Good one, i was just playing around with `input` validation colors. And missed removing those. Made the changes now 👍